includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -114,6 +110,39 @@ public static function get_detached_subscriptions( $limit = -1 ) {
 		return $detached_subscriptions;
 	}
 
+	/**
+	 * Checks if a subscription's payment method is detached from the customer.
+	 *
+	 * @param $subscription WC_Subscription The subscription object to check.
+	 * @return bool True if the payment method is detached, false otherwise.
+	 */
+	public static function is_subscription_payment_method_detached( $subscription ) {
+		$source_id = $subscription->get_meta( '_stripe_source_id' );
+		if ( ! $source_id ) {
+			return false;
+		}
+
+		$payment_method = WC_Stripe_API::get_payment_method( $source_id );
+		if ( is_wp_error( $payment_method ) ) {
```

Comment:
We should check here if the response has any error object as well (`isset( $payment_method->error )`

I coincidentally came across an old subscription on my local site while testing this PR. I see the subscription detached notice on the subscription edit page but there is no errors in my logs. When I logged the `$payment_method` variable to check, I found the following value-

`
{"error":{"code":"resource_missing","doc_url":"https:\/\/stripe.com\/docs\/error-codes\/resource-missing","message":"No such PaymentMethod: 'pm_id'","param":"payment_method","request_log_url":"https:\/\/dashboard.stripe.com\/acct_id\/test\/logs\/req_IUgF67UoEFgkAA?t=1751034251","type":"invalid_request_error"}}
`

Reply:
Thanks! Added the check in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4393/commits/f0b77614152b648bcb718a54fad8b168b9c1075f (included a fallback)